tagstudio/src/qt/thumbnailers/epub.py

```diff
@@ -0,0 +1,32 @@
+import zipfile
+from io import BytesIO
+from pathlib import Path
+
+import structlog
+from PIL import Image
+from src.qt.thumbnailers import ThumbnailBase
+
+logger = structlog.get_logger(__name__)
+
+
+class EpubThumbnail(ThumbnailBase):
+    EXTENSIONS = ("epub",)
+
+    @classmethod
+    def render(cls, filepath: Path, size) -> Image.Image | None:
+        try:
+            with zipfile.ZipFile(filepath, "r") as zip_file:
+                for file_name in zip_file.namelist():
+                    if file_name.lower().endswith(
+                        (".png", ".jpg", ".jpeg", ".gif", ".bmp", ".svg")
+                    ):
+                        image_data = zip_file.read(file_name)
+                        return Image.open(BytesIO(image_data))
```

Comment:
@cyanvoxel you might want to fix this in upstream as well - in the `ThumbRenderer` class you create `Image` instance for all images in a loop, but you use the last one only

https://github.com/TagStudioDev/TagStudio/blob/f6a1ca67835a8101e846619e6ea1ae40d227c425/tagstudio/src/qt/widgets/thumb_renderer.py#L657-L666